### PR TITLE
[Batch Infer] Fix different dataset types 

### PR DESCRIPTION
**Pull request by Eyal-Danieli**
This PR fix a bug in which the dataset is not provided as a `DateItem` object. 
In addition, we also add a new argument called `feature columns` that can be either received by the user or taken from the model artifact inputs. Note that these feature columns are being used when the provided dataset is from type list or numpy array. 

JIRA - https://jira.iguazeng.com/browse/ML-4477

### CODE REVIEW

File: batch_inference/batch_inference.py [hunk 1]
```diff
@@ -38,19 +38,22 @@
...
     Parse the given dataset into a DataFrame and drop the columns accordingly. In addition, the label columns will be
     parsed and validated as well.
 
-    :param dataset:       The dataset to train the model on.
-                          Can be either a list of lists, dict, URI or a FeatureVector.
-    :param label_columns: The target label(s) of the column(s) in the dataset. for Regression or
-                          Classification tasks.
-    :param drop_columns:  ``str`` / ``int`` or a list of ``str`` / ``int`` that represent the column names / indices to
-                          drop.
+    :param dataset:         The dataset to train the model on.
```
Comment (assaf758) [inline batch_inference/batch_inference.py:49]: shouldn't it be "The dataset the model was trained on" ?

**Review by assaf758 — COMMENTED** [review]
see inline.
you forgot to run function-to-item, no?

File: batch_inference/batch_inference.py (hunk 1 above)
Comment (Eyal-Danieli) [reply batch_inference/batch_inference.py]: Fixed, actually it should be a dataset that will be converted into a DataFrame (doesn't have to be something that is related to a training set at all).

**Comment by Eyal-Danieli** [comment]
> see inline. you forgot to run function-to-item, no?

right, fixed.

**Pull request merged**